common/flatpak-run.c

```diff
@@ -752,6 +752,35 @@ flatpak_run_apply_env_vars (FlatpakBwrap *bwrap, FlatpakContext *context)
     }
 }
 
+gboolean
+flatpak_run_apply_command (FlatpakBwrap *bwrap, FlatpakContext *context)
+{
+  GHashTableIter iter;
+  gpointer key, value;
+
+  g_hash_table_iter_init (&iter, context->command);
+  if (!g_hash_table_iter_next (&iter, &key, &value))
+    return FALSE;
+
+  const char *val = value;
+  gint argc;
+  g_auto(GStrv) argv = NULL;
+
+  if (!g_shell_parse_argv (val, &argc, &argv, NULL))
+    return FALSE;
```

Comment:
The error handling here is pretty bad: if the stored string is not in syntactically valid shell-like syntax, there's no error, not even a warning, and Flatpak will just silently fall back to the app's default. I think that's problematic for something as subtle as the shell-like syntax.

If it's stored internally as a `gchar **` instead, then that would avoid this problem.

Or if it's stored internally as a single string, I think this should at least emit the parse error as a `g_warning()`.